metaflow/client/core.py

```diff
@@ -25,13 +26,37 @@
     # python3
     import pickle
 
+# IMPORTANT NOTE ON THE CLIENT
+#
+# The client is not fully consistent particularly when you use the client to
+# query current runs. The previous version of the client accessed data directly
+# through whatever datastore existed (S3, etc) but could be partially
+# inconsistent because, it could, for example, return data from a previous
+# attempt while a new attempt was currently running (ie: depending on when you
+# use the client, you could get different values for the same data artifact).
+# This could happen because the client used the Metadata Service to retrieve the
+# artifacts to be read from the datastore and this information is only updated
+# at the *end* of an attempt.
+#
+# This current client goes through the datastore to access data artifacts
+# (well, it goes through the FileCache which uses the datastore). This means,
+# in particular, that it can render some things eventually consistent like the
+# attempt file and the DONE file (since it may try to read them before they
+# are written). This has *no* impact on Metaflow's execution since Metaflow
```

Comment:
Is this still valid after the latest update to S3 consistency?
https://aws.amazon.com/s3/consistency/

Reply:
No, I believe that we won't have as many issues. I'll update the comment.

Reply:
Done